decorators_library/decorators.py

```diff
@@ -1 +1,107 @@
-# implement your decorators here.
+import logging
+import signal
+import warnings
+
+from .exceptions import TimeoutError
+
+
+class debug(object):
+    
+    def __init__(self, logger=None):
+        self.logger = logger
+
+    def __call__(self, func):
+        if not self.logger:
+            logging.basicConfig()
+            log = logging.getLogger(func.__module__)
+            log.setLevel(logging.DEBUG)
+        else:
+            log = self.logger
+        
+        def func_wrapper(*args, **kwargs):
+            startmessage = 'Executing "{}" with params: {}, {}'.format(func.__name__, args, kwargs)
+            log.debug(startmessage)
+            results = func(*args, **kwargs)
+            endmessage = 'Finished "{}" execution with result: {}'.format(func.__name__, results)
+            log.debug(endmessage)
+            return results
+        return func_wrapper
+
+def timeout(time):
+    def wrapper(func):
+        def handler(signum, frame):
+            raise TimeoutError('Function call timed out')        
+        
+        def run_in_timer():
+            signal.signal(signal.SIGALRM, handler)
+            signal.alarm(time)
+            func()
```

Comment:
Don't you want to save and return the return value? There is no test for it, but your decorator should make a function loose its output, if their is any.